grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -356,13 +357,13 @@ class StockMovementController {
             throw new UnsupportedOperationException("${g.message(code: 'errors.noPermissions.label')}")
         }
         comment.properties = params
-        if (!comment.validate()) {
+        if (comment.validate()) {
             requisitionService.saveComment(comment)
```

Comment:
previously I meant that this save here would not be needed, because it is already a managed entity and calling `save()` explicitly is not needed, because it'd be saved at the end of the transaction.

It's not a "must change request", but just wanted to point that out, because in some cases (not this one, but I want to explain a potential problem somewhere), calling those `save` methods when they are not needed, might cause performance issues.

Assuming you have a managed entity that stores hundreds of items or something like that and for each change in the transaction you would call `entity.save()`, this would e.g. go through those items (and validate them for example) that many times, that you've called this save + 1 (because the save would be triggered at the end of the transaction, because it's the managed entity).

Not to be groundless - lately, thanks to removing that one, additional unnecessary `save` call, I've improved the performance of one request from **28 seconds to 3 seconds**.

Reply:
However comments don't have hundreds of items, so this should not be a problem here.

Reply:
this is why I gave an approval and wrote this:

> because in some cases (not this one, but I want to explain a potential problem somewhere),